### PR TITLE
test: Unify the serialisation tests

### PR DESCRIPTION
**Pull request by aborgna-q**
The tests all checked something different after the serialization rountrip.
This adds a check function to tests hugrs after the rondtrip.

Also drops the `metadata_hugr_ser` test, which was a duplicate from `weighted_hugr_ser`.

### CODE REVIEW

File: src/hugr/serialize.rs [hunk 1]
```diff
@@ -455,13 +441,6 @@ pub mod test {
         new_hugr.validate(&EMPTY_REG).unwrap_err();
         new_hugr.validate(&PRELUDE_REGISTRY).unwrap();
 
-        // Check the canonicalization works
```
[the remark this answers is not in the document]
Comment (aborgna-q) [reply src/hugr/serialize.rs:458]: Uhm, the code here was doing the same as the new function; checking that the canonicalization matches with the serialisation roundtrip.

It seems I didn't delete the serialisation code above.

**Review by ss2165 — APPROVED** [review]
